### ISSUE TITLE
Fix false positives for hashes in URLs in color-no-invalid-hex

### ISSUE THREAD

**Issue by ai** [stylelint/stylelint#4034]
> Clearly describe the bug

`fill=url(#a)`is parsing wrong by `color-no-invalid-hex`

> Which rule, if any, is the bug related to?

`color-no-invalid-hex`

> What CSS is needed to reproduce the bug?

```css
.invoice_logo {
  background-image: svg-load('../../icons/logo-color.svg', fill=url(#a));
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components"
  ],
  "rules": {
    "declaration-block-trailing-semicolon": null,
    "selector-list-comma-newline-after": null,
    "declaration-empty-line-before": null,
    "at-rule-empty-line-before": null,
    "no-descending-specificity": null,
    "no-duplicate-selectors": null,
    "rule-empty-line-before": null,
    "at-rule-no-unknown": [
      true,
      {
        "ignoreAtRules": [
          "util"
        ]
      }
    ]
  }
}
```

> Which version of stylelint are you using?

`10.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

```
stylelint blocks/**/*.sss globals/**/*.css controllers/*.js components/*.js pages/**/*.js pages/*.js popups/**/*.js popups/*.js layout/*.js uibook/*.js
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Standard CSS with [postcss-inline-svg](https://github.com/TrySound/postcss-inline-svg).

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

```
globals/invoice/invoice.css
 19:69  ✖  Unexpected invalid hex color "#a"   color-no-invalid-hex
```

**Comment by jeddy3** [comment]
@ai Thanks for the report and for using the template.

I can reproduce this.
